src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -70,14 +73,51 @@ public ClusterConnections(ClusterState clusterState, ClusterMembers clusterMembe
             _connectRetryStrategy = new RetryStrategy("connect to cluster", _clusterState.Options.Networking.ConnectionRetry, _clusterState.LoggerFactory);
 
             if (_clusterState.IsSmartRouting)
-            {
-                _smartRouting = true;
-                _connectAddresses = new ConnectAddresses(GetOrConnectAsync, _clusterState.LoggerFactory);
-            }
+                _connectMembers = ConnectMembers(memberConnectionQueue, _cancel.Token);
 
+            _clusterState.StateChanged += OnStateChanged;
+            
             HConsole.Configure(options => options.Set(this, x => x.SetPrefix("CCNX")));
         }
 
+        #region Connect Members
+
+        // background task that connect members
+        private async Task ConnectMembers(MemberConnectionQueue memberConnectionQueue, CancellationToken cancellationToken)
+        {
+            await foreach(var (member, token) in memberConnectionQueue.WithCancellation(cancellationToken))
+            {
+                var source = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, token);
```

Comment:
It's probably worse to wrap it into `using` statement: https://stackoverflow.com/q/61727174

Reply:
Yes, fixing.